api/main.py

```diff
@@ -259,6 +259,199 @@ def settle_btc_payments():
+                    time.sleep(60)
+                    continue
+
+                ln_payment_logs_util = LightningPaymentLogsUtil()
+
+                birdwatcher = get_birdwatcher()
+
+                for order in active_orders_with_lightning.all():
+                    app.logger.info(f"     ---- Processing order {order.id}... Order = {order}")
+
+                    invoices = order.invoices
+                    # app.logger.info(f"       -- Invoices for Order {order.id}... Invoice = {invoices}")
+
+                    for invoice in invoices:
+                        app.logger.info(f"       -- Invoice: {invoice.id} - {invoice.invoice}")
+
+                        # INCOMING PAYMENT
+                        incoming_payment_received = False
+
+                        if ln_payment_logs_util.check_incoming_payment(order.id, invoice.id, order.total):
+                            incoming_payment_received = True
+                            app.logger.info(f"Payment for order.id={order.id} WAS already recorded as received...")
+
+                        else:
+                            app.logger.info(f"Checking if payment for order id={order.id} is received...")
+
+                            incoming_invoice = incoming_invoices.get(invoice.invoice)
+
+                            if incoming_invoice:
+                                app.logger.info(f"Incoming invoice found: {incoming_invoice}")
+
+                                if incoming_invoice['is_paid']:
+                                    app.logger.info(f"   ******* _____ Invoice is paid _____ *******")
+
+                                    ln_payment_logs_util.add_incoming_payment(order.id, invoice.id, order.total)
+
+                                    incoming_payment_received = True;
+
+                                    # We don't mark the order as paid yet for the seller, but the buyer already paid,
+                                    # so we want him to have the order marked as paid so the QR dissapears from the screen
+                                    if not birdwatcher.send_dm(order.seller.parse_merchant_private_key(), order.buyer_public_key,
+                                        json.dumps({'id': order.uuid, 'type': 2, 'paid': True, 'shipped': False, 'message': f"Payment confirmed"})):
+                                        app.logger.info(f"     ********************** ERROR SENDING DM WITH TYPE=2, PAID=TRUE: {incoming_invoice}")
+
+                                else:
+                                    app.logger.info(f"   **** But not yet paid ****")
+
+                            else:
+                                app.logger.info(f"Payment for order.id={order.id} not received yet.")
+
+                        # OUTGOING PAYMENTS
+                        outgoing_payments_sent = True;
+
+                        if incoming_payment_received:
+                            payout_information = get_payout_information(order.seller_id)
```

Comment:
You don't need a function that gets the seller, converts it to dict, copies some fields in another dict, returns that here... you have `order.seller.lightning_address`, `order.seller.contribution_percent`... it's all here in the models.

Reply:
Yes, but I made this thinking in a future where we have to split the payment to 6 different persons. So here I'll be getting all those different persons  and calculating the percent for each one.